synthea/qpp_qicore.yaml

```diff
@@ -0,0 +1,171 @@
+      - profile: http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-patient
+        applicability: Patient
+      - profile: http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-encounter
+        applicability: Encounter
+      - profile: http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-condition
+        applicability: Condition
+      - profile: http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-observation
+        applicability: Observation
+      - profile: http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-procedure
+        applicability: Procedure
+      - profile: http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-medicationrequest
+        applicability: MedicationRequest
+      - profile: http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-immunization
+        applicability: Immunization
+      - profile: http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-careplan
+        applicability: CarePlan
+      - profile: http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-imagingstudy
+        applicability: ImagingStudy
+      - profile: http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-device
+        applicability: Device
+      - profile: http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-practitioner
+        applicability: Practitioner
+      - profile: http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-allergyintolerance
+        applicability: AllergyIntolerance
+      - profile: http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-claim
+        applicability: Claim
+      - profile: http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-coverage
+        applicability: Coverage
+      - profile: http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-servicerequest
+        applicability: ServiceRequest
+
+  - name: Set Missing Values
+    set_values:
+      - applicability: MedicationRequest
+        fields:
+          - location: MedicationRequest.doNotPerform
+            value: 'false'
+          - location: MedicationRequest.dispenseRequest.validityPeriod.start
+            value: $getField([MedicationRequest.authoredOn])
+  - name: Apply Script​
+    execute_script:
+      - apply_to: resource
+        function_name: addDiagnosis
+        resource_type: Encounter
+        function: |
+          function addDiagnosis(resource, bundle) {
+            const conditionEntry = bundle.entry.find(e => e.resource?.resourceType === 'Condition' && e.resource?.encounter?.reference === `urn:uuid:${resource.id}`);
+            if (conditionEntry){
+              resource.diagnosis = [{
+                condition:{
+                  reference: `Condition/${conditionEntry.resource.id}`
+                }
+              }];
+            }
+          }
+  - name: Create Resources
+    create_resource:
+      - resourceType: ServiceRequest
+        based_on:
+          resource: Procedure.performed.ofType(Period) # handle value setting for period choice type
```

Comment:
This can be cleaned up now - since Lauren originally wrote the earlier mapping file I added support for `if` -- see https://github.com/synthetichealth/synthea/pull/1525
Basically instead of duplicated create_resource entries based on different possible types of Procedure.performed, you can collapse that down to one overall entry for Procedure, and just different fields with an `if` that distinguishes between the possible field types

Reply:
OH awesome, thanks Dylan!!